File: src/test/resources/yield/ThrowsSuper.cs
using System.Collections;

namespace yield {
    public sealed class ThrowsSuper {
        public class Base {
            public virtual IEnumerator test() {
                yield break;
            }
        }

        public class Child : yield.ThrowsSuper.Base {
            public override IEnumerator test() {
                yield return base.test();
                yield break;
            }
        }
    }
}

File: src/test/resources/yield/ThrowsSuper.java
package yield;

public final class ThrowsSuper {

    /**
     * @sharpen.ignore
     * @sharpen.yield
     */
    public static class Marker extends Exception {
    }

    public static class Base {
        public void test() throws Marker {
            return;
        }
    }

    public static class Child extends Base {
        public void test() throws Marker {
            super.test();
            return;
        }
    }
}
